llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp

```diff
@@ -1296,9 +1438,16 @@ bool AddressSanitizer::isInterestingAlloca(const AllocaInst &AI) {
 bool AddressSanitizer::ignoreAccess(Instruction *Inst, Value *Ptr) {
   // Instrument accesses from different address spaces only for AMDGPU.
   Type *PtrTy = cast<PointerType>(Ptr->getType()->getScalarType());
-  if (PtrTy->getPointerAddressSpace() != 0 &&
-      !(TargetTriple.isAMDGPU() && !isUnsupportedAMDGPUAddrspace(Ptr)))
+  if (TargetTriple.isSPIR()) {
+    // SPIR has its own rules to filter the instrument accesses, it also filters
+    // some AS=0 instructions
+    if (isUnsupportedSPIRAddrspace(Ptr, Inst->getFunction())) {
+      return true;
+    }
```

Comment:
```suggestion
    if (isUnsupportedSPIRAddrspace(Ptr, Inst->getFunction()))
      return true;
```

Nit, see [LLVM Coding Standards](https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements)

Reply:
Done